modules/visual_script/editor/visual_script_editor.cpp

```diff
@@ -3487,106 +3480,105 @@ void VisualScriptEditor::_selected_connect_node(const String &p_text, const Stri
+			} else if (property_path_slice == "Basis") {
 				n->set_basic_type(Variant::BASIS);
-			} else if (property_path[0] == "Transform3D") {
+			} else if (property_path_slice == "Transform3D") {
 				n->set_basic_type(Variant::TRANSFORM3D);
-			} else if (property_path[0] == "Color") {
+			} else if (property_path_slice == "Color") {
 				n->set_basic_type(Variant::COLOR);
-			} else if (property_path[0] == "RID") {
+			} else if (property_path_slice == "RID") {
 				n->set_basic_type(Variant::RID);
-			} else if (property_path[0] == "Object") {
+			} else if (property_path_slice == "Object") {
 				n->set_basic_type(Variant::OBJECT);
-			} else if (property_path[0] == "Callable") {
+			} else if (property_path_slice == "Callable") {
 				n->set_basic_type(Variant::CALLABLE);
-			} else if (property_path[0] == "Signal") {
+			} else if (property_path_slice == "Signal") {
 				n->set_basic_type(Variant::SIGNAL);
-			} else if (property_path[0] == "StringName") {
+			} else if (property_path_slice == "StringName") {
 				n->set_basic_type(Variant::STRING_NAME);
-			} else if (property_path[0] == "NodePath") {
+			} else if (property_path_slice == "NodePath") {
 				n->set_basic_type(Variant::NODE_PATH);
-			} else if (property_path[0] == "Dictionary") {
+			} else if (property_path_slice == "Dictionary") {
 				n->set_basic_type(Variant::DICTIONARY);
-			} else if (property_path[0] == "Array") {
+			} else if (property_path_slice == "Array") {
 				n->set_basic_type(Variant::ARRAY);
-			} else if (property_path[0] == "PackedByteArray") {
+			} else if (property_path_slice == "PackedByteArray") {
 				n->set_basic_type(Variant::PACKED_BYTE_ARRAY);
-			} else if (property_path[0] == "PackedInt32Array") {
+			} else if (property_path_slice == "PackedInt32Array") {
 				n->set_basic_type(Variant::PACKED_INT32_ARRAY);
-			} else if (property_path[0] == "PackedInt64Array") {
+			} else if (property_path_slice == "PackedInt64Array") {
 				n->set_basic_type(Variant::PACKED_INT64_ARRAY);
-			} else if (property_path[0] == "PackedFloat32Array") {
+			} else if (property_path_slice == "PackedFloat32Array") {
 				n->set_basic_type(Variant::PACKED_FLOAT32_ARRAY);
-			} else if (property_path[0] == "PackedStringArray") {
+			} else if (property_path_slice == "PackedStringArray") {
 				n->set_basic_type(Variant::PACKED_STRING_ARRAY);
-			} else if (property_path[0] == "PackedVector2Array") {
+			} else if (property_path_slice == "PackedVector2Array") {
 				n->set_basic_type(Variant::PACKED_VECTOR2_ARRAY);
-			} else if (property_path[0] == "PackedVector3Array") {
+			} else if (property_path_slice == "PackedVector3Array") {
 				n->set_basic_type(Variant::PACKED_VECTOR3_ARRAY);
-			} else if (property_path[0] == "PackedColorArray") {
+			} else if (property_path_slice == "PackedColorArray") {
 				n->set_basic_type(Variant::PACKED_COLOR_ARRAY);
 			}
-			n->set_basic_type_constant(property_path[1]);
+			n->set_basic_type_constant(p_text.get_slice(":", 1));
```

Comment:
I would extract this together with the `base_type` before the huge list of `if`/`else`, so the code is more self-explanatory:
```
const String base_type = p_text.get_slice(":", 0);
const String class_constant = p_text.get_slice(":", 1);
...
```